game/src/game/player.rs

```diff
@@ -61,6 +132,563 @@ fn setup_player(q: Query<(&Transform, Entity), Added<PlayerBlueprint>>, mut comm
             },
             animation: Default::default(),
         },));
+        println!("player spawned")
+    }
+}
+
+struct PlayerStatePlugin;
+
+impl Plugin for PlayerStatePlugin {
+    fn build(&self, app: &mut App) {
+        app.add_systems(
+            Update,
+            (
+                // test_trans_int,
+                // transition::<Idle, Grounded>,
+                transition::<Falling, Grounded>,
+                transition::<Grounded, Falling>,
+                // transition::<Falling, Jumping>,
+                transition::<Falling, Running>,
+                transition::<Grounded, Jumping>,
+                transition::<Jumping, Falling>,
+                transition::<Idle, Running>,
+                transition::<Running, Idle>,
+                apply_deferred,
```

Comment:
What is this apply_deferred doing here? As is, it is unordered relative to other systems so it doesn't really accomplish anything meaningful.

Reply:
I was going to add the transition systems to a set and then order the animation systems after it to make sure they ran post transition every frame but didn't get there yet.